cpp/ql/lib/semmle/code/cpp/ir/dataflow/MustFlow.qll

```diff
@@ -48,30 +47,30 @@ abstract class MustFlowConfiguration extends string {
    * included in the module `PathGraph`.
    */
   final predicate hasFlowPath(MustFlowPathNode source, MustFlowPathSink sink) {
-    this.isSource(source.getNode()) and
+    this.isSource(source.getInstruction()) and
     source.getASuccessor+() = sink
   }
 }
 
 /** Holds if `node` flows from a source. */
 pragma[nomagic]
-private predicate flowsFromSource(DataFlow::Node node, MustFlowConfiguration config) {
+private predicate flowsFromSource(Instruction node, MustFlowConfiguration config) {
   config.isSource(node)
   or
-  exists(DataFlow::Node mid |
+  exists(Instruction mid |
     step(mid, node, config) and
     flowsFromSource(mid, pragma[only_bind_into](config))
   )
 }
 
 /** Holds if `node` flows to a sink. */
 pragma[nomagic]
-private predicate flowsToSink(DataFlow::Node node, MustFlowConfiguration config) {
+private predicate flowsToSink(Instruction node, MustFlowConfiguration config) {
   flowsFromSource(node, pragma[only_bind_into](config)) and
   (
-    config.isSink(node)
+    config.isSink(node.getAUse())
```

Comment:
Is it guaranteed that there still will be a use?

Reply:
The sink is defined to be an operand. So if there's a sink, then it's guaranteed to be an operand, and that operand will be a use of some instruction (i.e., the instruction currently named `node`). Does that make sense?

Reply:
I guess my question is: is the result of an instruction (if it has one) always guaranteed to be used?

Reply:
I don't think so, no.

Reply:
But those cases weren't captured before either?

Reply:
Correct. If the def-use isn't captured explicitly in the IR, then it was never captured by the `MustFlow` library.

Reply:
That makes sense. Thanks.